mesonbuild/compilers/fortran.py

```diff
@@ -0,0 +1,281 @@
+
+
+class SunFortranCompiler(FortranCompiler):
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'sun'
+
+    def get_dependency_gen_args(self, outtarget, outfile):
+        return ['-fpp']
+
+    def get_always_args(self):
+        return []
+
+    def get_warn_args(self, level):
+        return []
+
+    def get_module_outdir_args(self, path):
+        return ['-moddir=' + path]
+
+
+class IntelFortranCompiler(IntelCompiler, FortranCompiler):
+    std_warn_args = ['-warn', 'all']
+
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        self.file_suffixes = ('f90', 'f', 'for', 'ftn', 'fpp')
+        FortranCompiler.__init__(self, exelist, version, is_cross, exe_wrapper)
+        # FIXME: Add support for OS X and Windows in detect_fortran_compiler so
+        # we are sent the type of compiler
+        IntelCompiler.__init__(self, ICC_STANDARD)
+        self.id = 'intel'
+
+    def get_module_outdir_args(self, path):
+        return ['-module', path]
+
+    def get_warn_args(self, level):
+        return IntelFortranCompiler.std_warn_args
+
+
+class PathScaleFortranCompiler(FortranCompiler):
+    std_warn_args = ['-fullwarn']
+
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'pathscale'
+
+    def get_module_outdir_args(self, path):
+        return ['-module', path]
+
+    def get_std_warn_args(self, level):
+        return PathScaleFortranCompiler.std_warn_args
+
+
+class PGIFortranCompiler(FortranCompiler):
+    std_warn_args = ['-Minform=inform']
+
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'pgi'
+
+    def get_module_outdir_args(self, path):
+        return ['-module', path]
+
+    def get_warn_args(self, level):
+        return PGIFortranCompiler.std_warn_args
+
+    def get_no_warn_args(self):
+        return ['-silent']
+
+
+class Open64FortranCompiler(FortranCompiler):
+    std_warn_args = ['-fullwarn']
+
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'open64'
+
+    def get_module_outdir_args(self, path):
+        return ['-module', path]
+
+    def get_warn_args(self, level):
+        return Open64FortranCompiler.std_warn_args
+
+
+class NAGFortranCompiler(FortranCompiler):
+    std_warn_args = []
+
+    def __init__(self, exelist, version, is_cross, exe_wrapper=None):
+        super().__init__(exelist, version, is_cross, exe_wrapper=None)
+        self.id = 'nagfor'
+
+    def get_module_outdir_args(self, path):
+        return ['-mdir', path]
+
+    def get_warn_args(self, level):
+        return NAGFortranCompiler.std_warn_args
+
```

Comment:
__[flake8]__
* [W391] blank line at end of file

<!-- Comment by [SideCI](https://sideci.com) -->